File: src/test/java/homework8/expandtesting/TestExpandtesting.java
package homework8.expandtesting;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.time.Duration;

public class TestExpandtesting extends BaseTestExpandtesting {

    @Test
    void testSelectedData() {
        driver.get(BASEURL + "/webpark");

        Select dropDownMenu = new Select(driver.findElement(By.id("parkingLot")));
        dropDownMenu.selectByVisibleText("Short-Term Parking");


        WebElement entryDate = driver.findElement(By.id("entryDate"));
        WebElement entryTime = driver.findElement(By.id("entryTime"));
        WebElement exitDate = driver.findElement(By.id("exitDate"));
        WebElement exitTime = driver.findElement(By.id("exitTime"));

        entryDate.clear();
        entryDate.sendKeys("2023-08-14");
        entryTime.clear();
        entryTime.sendKeys("02:00");
        exitDate.clear();
        exitDate.sendKeys("2023-08-16");
        exitTime.clear();
        exitTime.sendKeys("05:00");

        driver.findElement(By.id("calculateCost")).click();

        int result = Integer.parseInt(new WebDriverWait(driver, Duration.ofSeconds(3)).until(ExpectedConditions.presenceOfElementLocated(By.id("resultValue"))).getText().split(".00€")[0]);
        Assert.assertEquals(result, getPrice("2023-08-14", "02:00", "2023-08-16", "05:00"));
    }
}

File: homework8/expandtesting/TestExpandtesting.py
from time import sleep

from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

from homework8.expandtesting.BaseTestExpandtesting import BaseTestExpandtesting


class TestExpandtesting(BaseTestExpandtesting):

    def test_selected_date(self):
        self.driver.get(self.BASE_URL + "/webpark")

        entry_date = self.driver.find_element(By.ID, "entryDate")
        entry_time = self.driver.find_element(By.ID, "entryTime")
        exit_date = self.driver.find_element(By.ID, "exitDate")
        exit_time = self.driver.find_element(By.ID, "exitTime")

        entry_date.clear()
        entry_date.send_keys("2023-08-14")
        entry_time.clear()
        entry_time.send_keys("02:00")
        exit_date.clear()
        exit_date.send_keys("2023-08-16")
        exit_time.clear()
        exit_time.send_keys("05:00")

        self.driver.find_element(By.ID, "calculateCost").click()

        result = WebDriverWait(self.driver, 5).until(EC.presence_of_element_located((By.ID, "resultValue"))).text.split(".00€")[0]

        assert int(result) == int(self.price("2023-08-14 02:00", "2023-08-16 05:00"))


